### ISSUE TITLE
On Linux a Seg fault/ core dump occurs when loading Geos via sDNAPrepare

### ISSUE THREAD

**Issue by JamesParrott** [fiftysevendegreesofrad/sdna_plus#77]
In the last smoke test (`python -u `prepare_test_new.py`):

https://github.com/fiftysevendegreesofrad/sdna_plus/actions/runs/9584489142/job/26428459946

This will also cause a regression test failure of the correct_outprep.txt test (due to issue #65 a different error occurs first, shortly after the line "4-link subsystem contains link with id = 8028")

```

link unlink import test
Using xytolerance=0, ztolerance=0
before
net - POLYLINEZ (10 items)
[('ID', 'ID', 'INT'), ('sametext', 'sametext', 'STR'), ('difftext', 'difftext', 'STR'), ('start_gs', 'start_gs', 'FLOAT'), ('end_gs', 'end_gs', 'FLOAT'), ('weight', 'weight', 'FLOAT'), ('cost', 'cost', 'FLOAT'), ('island', 'island', 'FLOAT')]
[('ID', 0), ('sametext', '1'), ('difftext', '0'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((0,0,0),(1,1,0)))
[('ID', 1), ('sametext', '1'), ('difftext', '1'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((1,0,0),(0,1,0)))
[('ID', 2), ('sametext', '1'), ('difftext', '2'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((1,1,0),(2,1,0)))
[('ID', 3), ('sametext', '1'), ('difftext', '3'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((2,0,0),(2,2,0)))
[('ID', 4), ('sametext', '1'), ('difftext', '4'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((10,0,0),(11,1,0)))
[('ID', 5), ('sametext', '1'), ('difftext', '5'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((11,0,0),(10,1,0)))
[('ID', 6), ('sametext', '1'), ('difftext', '6'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((11,1,0),(12,1,0)))
[('ID', 7), ('sametext', '1'), ('difftext', '7'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((12,0,0),(12,2,0)))
[('ID', 8), ('sametext', '1'), ('difftext', '8'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((3,3,0),(6,6,0)))
[('ID', 9), ('sametext', '1'), ('difftext', '9'), ('start_gs', 0.0), ('end_gs', 0.0), ('weight', 1.0), ('cost', 1.0), ('island', 0.0)]  (((4,4,0),(5,5,0)))
/home/runner/work/_temp/1e1ec8bd-d94b-498d-85ae-ff71fa444658.sh: line 1: 11890 Segmentation fault      (core dumped) python -u prepare_test_new.py
Error: Process completed with exit code 139.
```

**Comment by JamesParrott** [comment]
I have not been able to dynamically link to a .so, from within a .so on Ubuntu at all, without causing a seg fault. Minimal example:
https://github.com/JamesParrott/So_from_so

**Comment by JamesParrott** [comment]
Fixed by PR #78
